mypy/plugins/common.py

```diff
@@ -155,8 +157,24 @@ def add_method_to_class(
 
 
 def deserialize_and_fixup_type(
-    data: Union[str, JsonDict], api: SemanticAnalyzerPluginInterface
+    data: Union[str, JsonDict],
+    api: Union[SemanticAnalyzerPluginInterface, CheckerPluginInterface]
 ) -> Type:
     typ = deserialize_type(data)
     typ.accept(TypeFixer(api.modules, allow_missing=False))
     return typ
+
+
+def get_anonymous_typeddict_type(api: CheckerPluginInterface) -> Instance:
+    for type_fullname in TPDICT_FB_NAMES:
+        try:
+            return api.named_generic_type(type_fullname, [])
```

Comment:
This is actually not the best method for this purpose, it will fail if `typing_extensions` is not imported in the current module. Instead, you need to use something like `named_type_or_none()`, see e.g. fallback code in `build_typeddict_typeinfo()` in `semanal_typeddict.py`. It is however not available in `TypeChecker`, so maybe you can try to copy it from `SemanticAnalyzer`.